src/mahoji/commands/buy.ts

```diff
@@ -82,6 +88,37 @@ export const buyCommand: OSBMahojiCommand = {
 				content: `Removed ${cost} from your bank.`
 			};
 		}
+		if (stringMatches(name, 'fossil island notes')) {
+			const cost = new Bank().add('Numulite', 300);
+			if (user.minionIsBusy) {
+				return 'Your minion is busy.';
+			}
+			if (user.QP < 3) {
+				return 'You need 3qp to reach the stone chest';
+			}
+			if (!user.owns(cost)) {
+				return "You don't have enough Numulite.";
+			}
+			const allItemsOwnedBank = user.allItemsOwned();
+			let loot = new Bank();
+			if (fossilIslandNotesCL.every(page => allItemsOwnedBank.has(page))) {
+				const outPage = getOSItem(randArrItem(fossilIslandNotesCL));
+				loot.add(outPage.id);
+			} else {
+				const filteredPages = fossilIslandNotesCL.filter(page => !allItemsOwnedBank.has(page));
+				const outPage = getOSItem(randArrItem(filteredPages));
+				loot.add(outPage.id);
+			}
+			await handleMahojiConfirmation(
+				interaction,
+				`${user}, please confirm that you want to buy a Fossil Island note for: ${cost}.`
+			);
+
+			await transactItems({ userID: user.id, itemsToRemove: cost, itemsToAdd: loot, collectionLog: true });
+
+			return `You purchased ${loot}.`;
+		}
+
```

Comment:
This should be atleast put into a abstracted command file, and imported in here.

Reply:
I used the same method as the kitten purchase above, would you want that moving also?

Reply:
I did the same as the kitten purchase above, do you want that moving also?